### DISCUSSION TITLE
Add "change" parameter to TextEdit's text_changed signal

### DISCUSSION THREAD

**Discussion by InkONat** [godotengine/godot-proposals#7584] [GUI]
Add a parameter called "change" to the text_changed signal from TextEdit being the added/removed character from the text in TextEdit

Example:

![IMG_20230830_082354](https://github.com/godotengine/godot-proposals/assets/61206149/e3163e4b-f130-4572-98c0-9c91a01c2213)

If you change the text on TextEdit from "text" to "text!" it will print `!` and if you change from "text!" to "text" it will also print `!`

(Sorry if this is a bad proposal its my first time making a feature proposal)

**Comment by timothyqiu** [comment]
How could the user know whether `change` is added or removed?

e.g. Changing from `hello` to `hellohello` and changing from `hellohellohello` to `hellohello` both leave you with the same `text` and `change`.

**Reply by InkONat** [reply]
We would need a second parameter for that I think

**Reply by BeayemX** [reply]
Could you provide more info regarding what you are trying to do?

For your proposed solution the signal could have both values as parameters
`_on_text_changed(current_text, previous_text)`, so that you can create the difference by yourself and see which characters were added or removed.

To achieve this for now, you could just store the text in a member variable and when handling a text change via the signal you would have the `previous_text`as a member variable and the new text as the current `.text`.

**Reply by InkONat** [reply]
Yea that's a good idea
